probatus/missing_values/imputation.py

```diff
@@ -0,0 +1,384 @@
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+from probatus.utils import (
+    preprocess_data,
+    preprocess_labels,
+    BaseFitComputePlotClass,
+    get_single_scorer,
+)
+from sklearn.model_selection import cross_val_score
+from sklearn.pipeline import Pipeline
+from sklearn.impute import SimpleImputer
+from sklearn.compose import ColumnTransformer
+from sklearn.preprocessing import OneHotEncoder
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+
+class ImputationSelector(BaseFitComputePlotClass):
+    """
+    Comparison of various imputation strategies that can be used for imputing
+    missing values.
+    The aim of this class is to present the model performance based on imputation
+    strategies and a choosen model.
+    For models like XGBoost & LighGBM which have capabilities to handle missing values by default
+    the model performance with no imputation will be shown as well.
+    The missing values categorical features are imputed with the value `missing` and an missing indicator is
+    added.
+
+    Example usage.
+    ```python
+
+    #Import the class
+
+    import pandas as pd
+    import numpy as np
+    import matplotlib.pyplot as plt
+    from probatus.missing_values.imputation import ImputationSelector
+    from probatus.utils.missing_helpers import generate_MCAR
+    from sklearn.linear_model import LogisticRegression
+    from sklearn.experimental import enable_iterative_imputer
+    from sklearn.impute import KNNImputer,SimpleImputer,IterativeImputer
+    from sklearn.datasets import make_classification
+
+    #Create data with missing values.
+    n_features = 10
+    X,y = make_classification(n_samples=1000,n_features=n_features,random_state=123,class_sep=0.3)
+    X = pd.DataFrame(X, columns=["f_"+str(i) for i in range(0,n_features)])
+    X_missing = generate_MCAR(X,missing=0.2)
+
+    #Create the strategies.
+    strategies = {
+       'Simple Median Imputer' : SimpleImputer(strategy='median',add_indicator=True),
+       'Simple Mean Imputer' : SimpleImputer(strategy='mean',add_indicator=True),
+       'Iterative Imputer'  : IterativeImputer(add_indicator=True,n_nearest_features=5,
+       sample_posterior=True),
+       'KNN' : KNNImputer(n_neighbors=3)}
+    #Create a classifier.
+    clf = LogisticRegression()
+    #Create the comparision of the imputation strategies.
+    cmp = ImputationSelector(
+        clf=clf,
+        strategies=strategies,
+        cv=5,
+        model_na_support=False)
+
+    cmp.fit_compute(X_missing,y)
+    #Plot the results.
+    cmp.plot()
+
+    ```
+    <img src="../img/imputation_comparision.png" width="500" />
+
+    """
+
+    def __init__(
+        self,
+        clf,
+        strategies,
+        scoring="roc_auc",
```

Comment:
Let's do first cv parameter and then scoring.
To be fully consistent with other classes, see: https://github.com/ing-bank/probatus/blob/main/probatus/feature_elimination/feature_elimination.py

Reply:
Done.